website/static/js/mailingListTreebeard.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+var $ = require('jquery');
+var m = require('mithril');
+var Treebeard = require('treebeard');
+var $osf = require('js/osfHelpers');
+var $tb = require('js/treebeardHelpers');
+var projectSettingsTreebeardBase = require('js/projectSettingsTreebeardBase');
+
+function subscribe(item, notification_type, reload) {
+    var id = item.data.node.id;
+    var url = '/api/v1/project/' + id + '/mailing_list/';
```

Comment:
I'm actually a bit surprised that we're adding new endpoints/ functionality to APIv1 at this point. Can you please clarify why it's not being written from scratch for APIv2?

(since presumably we'd expect a rewrite anyway)

Fitz was compiling a list of api v1 endpoints that are missing from apiv2. I don't think apiv2 should block this PR, but we should be sure to get these new endpoints added to his list.